packages/react-components/react-button/stories/Button/ButtonActionState.stories.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { makeStyles, Button, Tooltip } from '@fluentui/react-components';
+import { ButtonActionState } from '../../src/components/Button/Button.types';
+import { Cloud20Regular, CloudCheckmark20Regular, CloudSync20Regular, SaveRegular } from '@fluentui/react-icons';
+
+const useStyles = makeStyles({
+  wrapper: {
+    alignItems: 'center',
+    columnGap: '15px',
+    display: 'flex',
+  },
+});
+
+export const ActionState = () => {
+  const styles = useStyles();
+
+  const [buttonState, setButtonState] = React.useState<ButtonActionState>('none');
+
+  function changeState() {
+    setButtonState('inprogress');
+    setTimeout(() => {
+      setButtonState('complete');
+      setTimeout(() => {
+        setButtonState('none');
+      }, 3000);
+    }, 3000);
+  }
```

Comment:
Important to invalidate the timeout here:

```suggestion
  const stateTimeout = React.useRef(null);

  function changeState() {
    clearTimeout(stateTimeout.current);

    setButtonState('inprogress');
    stateTimeout.current = setTimeout(() => {
      setButtonState('complete');

      stateTimeout.current = setTimeout(() => {
        setButtonState('none');
      }, 3000);
    }, 3000);
  }
```